app/helpers/ZenifyFixtures/Alice/AliceLoader.php

```diff
@@ -20,86 +20,65 @@
 final class AliceLoader implements AliceLoaderInterface
 {
 
-	/**
-	 * @var Loader
-	 */
-	private $aliceLoader;
-
-	/**
-	 * @var EntityManagerInterface
-	 */
-	private $entityManager;
-
-
-	public function __construct(Loader $aliceLoader, EntityManagerInterface $entityManager)
-	{
-		$this->aliceLoader = $aliceLoader;
-		$this->entityManager = $entityManager;
-	}
-
-
-	/**
-	 * @param string|array $sources
-	 * @return object[]
-	 */
-	public function load($sources) : array
-	{
-		if ( ! is_array($sources)) {
-			$sources = [$sources];
-		}
-
-		$entities = [];
-		foreach ($sources as $source) {
-			$newEntities = $this->loadEntitiesFromSource($source);
-			$entities = array_merge($entities, $newEntities);
-		}
-
-		$this->entityManager->flush();
-
-		return $entities;
-	}
-
-
-	/**
-	 * @param mixed $source
-	 * @return object[]
-	 */
-	private function loadEntitiesFromSource($source) : array
-	{
-		if (is_dir($source)) {
-			return $this->loadFromDirectory($source);
-
-		} elseif (is_file($source)) {
-			return $this->loadFromFile($source);
-		}
-
-		throw new MissingSourceException(
-			sprintf('Source "%s" was not found.', $source)
-		);
-	}
-
-
-	/**
-	 * @return object[]
-	 */
-	private function loadFromFile(string $path) : array
-	{
-		$entities = $this->aliceLoader->load($path);
-		foreach ($entities as $entity) {
-			$this->entityManager->persist($entity);
-		}
-		return $entities;
-	}
-
-
-	private function loadFromDirectory(string $path) : array
-	{
-		$files = [];
-		foreach (Finder::find('*.neon', '*.yaml', '*.yml')->from($path) as $file) {
-			/** @var SplFileInfo $file */
-			$files[] = $file->getPathname();
-		}
-		return $this->load($files);
-	}
-
+    /**
+     * @var CustomNativeLoader
+     */
+    private $aliceLoader;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+
+    public function __construct(CustomNativeLoader $aliceLoader, EntityManagerInterface $entityManager)
+    {
+        $this->aliceLoader = $aliceLoader;
+        $this->entityManager = $entityManager;
+    }
+
+
+    /**
+     * @param string|array $sources
+     * @return object[]
+     */
+    public function load($sources): array
+    {
+        if (!is_array($sources) && is_dir($sources)) {
+            $sources = $this->getFilesFromDirectory($sources);
+        } else if (!is_array($sources)) {
+            $sources = [$sources];
+        }
+
+        $this->checkExistence($sources);
+
+        $entities = $this->aliceLoader->loadFiles($sources)->getObjects();
+        foreach ($entities as $entity) {
+            $this->entityManager->persist($entity);
+        }
+        $this->entityManager->flush();
+
+        return $entities;
+    }
+
+    private function checkExistence(array $sources)
+    {
+        foreach ($sources as $source) {
+            if (!is_file($source)) {
```

Comment:
Maybe we should also add is_readable() check? Just a thought.

Reply:
yeah, why not